invenio_deposit/api.py

```diff
@@ -341,7 +341,7 @@ def publish(self, pid=None, id_=None):
             self._publish_new(id_=id_)
         else:  # Update after edit
             record = self._publish_edited()
-            record.commit()
+            record.commit(**kwargs)
```

Comment:
For consistency reasons, I would say that all `kwargs` should go to method binded to `self`. In this specific case we can add new argument `record_kwargs`, `record_ctx`, or similar. Then the code should be something like `record.commit(**record_kwags or {})`.